packages/foundation/docs/themes.mdx

```diff
@@ -1,30 +1,232 @@
+UX design to be poor for the user experience.
+
+If you use the component `UiView` then it sets automatically out of the box the theme context, so you SHOULD wrap your entire app
+in a UiView, you can get more information about this component in the [UiView docs](https://uireact.io/packages-view-docs-view).
+
+TL;DR;
+
+```
+import { @UiView } from '@uireact/view';
+import { DefaultTheme, ThemeColor } from '@uireact/foundation';
+
+export const App = () => (
+  <UiView theme={DefaultTheme} selectedTheme={ThemeColor.light}>
+    ... {Your App} ...
+  </Uiview>
+);
+
+```
+
+If you don't want to use the view component then you will have to manually set the ThemeContext provided from the foundation package:
+
+```
+import { ThemeContext, DefaultTheme, ThemeColor } from '@uireact/foundation';
+
+export const App = () => (
+  <ThemeContext.Provider value={{ theme: DefaultTheme, selectedTheme: ThemeColor.light }}>
+    ... {Your App} ...
+  </ThemeContext.Provider>
+);
+
+```
+
+## Structure 🧱
+
+Each theme has to meet the structure of the [Theme](https://github.com/inavac182/uireact/blob/main/packages/foundation/src/types/themes/theme.ts) type.
+
+```
+export const DefaultTheme: Theme = {
+  name: 'UIReactDefaultTheme',
+    ▶︎ A name for the theme
+  dark: { ... },
+    ▶︎ A set of colors for the dark coloration
+  light: { ... },
+    ▶︎ A set of colors for the light coloration
+  sizes: { ...},
+    ▶︎ The set of sizes you want to use for fonts / headings / icons
+  texts: { ... },
+    ▶︎ The configuration for texts such as fonts
+  spacing: { ... },
+    ▶︎ Levels of spacing with its px value
+};
+
+```
+
+## Colorations ✨🌓
+
+Our take on UX design nowadays is that you should be ready to provide with 2 colorations in your UI, dark and light, this to help
+the UX of your applications when used during sunny days 🌞 or during night time 🌙
+
+Each coloration has to include these categories:
+
+```
+{
+  backgrounds: { ... },
+    ▶︎ Used for background or components that are meant to wrap content such as the view or cards.
+  fonts: { ... },
+    ▶︎ Used for texts, headings, labels, everything that is text related.
+  primary: { ... },
+    ▶︎ This is the primary color in your application and will used accross the board.
+  secondary: { ... },
+    ▶︎ The secondary color SHOULD contract very well with the primary, as is intended to be used to separate contents.
+  tertiary: { ... },
+    ▶︎ The tertiary color is like a sparkle here and there, not commonly used but meant to be very visible when is used.
+  positive: { ... },
+    ▶︎ Commonly used to show a reaction to an action that was successfully performed, e.g. A form submition.
+  negative: { ... },
+    ▶︎ Used to show a fallback action, meaning the less ideal action that we intend our users to do
+  error: { ... },
+    ▶︎ As its name says, this is used to show bright error messages, e.g. A form submittion that failed to be validated.
+}
+
+```
```

Comment:
In this part I'm not sure I understand if Coloration is part of the `Theme` or something different, it is indicating that I need to `provide with 2 colorations` but I'm a bit unsure where and how